internal/olm/client/client.go

```diff
@@ -48,6 +48,22 @@ var ErrOLMNotInstalled = errors.New("no existing installation found")
 
 var Scheme = scheme.Scheme
 
+// custom error struct to capture deployment errors
+// while verifying CSV installs.
+type deploymentError struct {
+	depName string
+	issue   string
+}
+type deploymentVerifyError []deploymentError
+
+func (e deploymentVerifyError) Error() string {
+	var sb strings.Builder
+	for _, i := range e {
+		sb.WriteString(i.depName + " has errors: \n" + i.issue + "\n")
+	}
+	return sb.String()
+}
```

Comment:
We have an implementation to return a list of errors in the project. See: https://github.com/operator-framework/operator-sdk/blob/16403cbe1c152d3ee7f21987aacb88444fc9cf80/internal/cmd/operator-sdk/bundle/internal/result.go

IMO we should move it for internal/util and start to use it here as well. 

Note that more than one place has been needing address the same need (bundle validate, scorecard and now bundle run) and instead of we centralize it o keep ALL using the same we are creating a new implementation in each place which reduces the maintainability. 

WDYT @jmrodri @estroz

Reply:
sounds good to me. I think we have a bit too much stuff like that squirreled away in internal packages instead of in one unified util package.

Reply:
I disagree that `Result` should be used, see https://github.com/operator-framework/operator-sdk/pull/3908#discussion_r496232955.

Having one unified util package has its own drawbacks. There's a reason we have multiple internal packages, although I agree some stuff can be percolated up into parent internal packages.

Reply:
if we are able to change it without need to bump a new version I am ok with. I raise an issue for that:  
https://github.com/operator-framework/operator-sdk/issues/3966